File: org.palladiosimulator.blockchainsystems.plugin/src/org/palladiosimulator/blockchainsystems/plugin/creation/connectedsubgraphnetwork/ConnectedSubgraphNetworkCreationResult.java
package org.palladiosimulator.blockchainsystems.plugin.creation.connectedsubgraphnetwork;

import java.util.HashMap;

import org.palladiosimulator.blockchainsystems.core.system.abstractions.P2PNetwork;
import org.palladiosimulator.blockchainsystems.core.system.abstractions.P2PNetworkCreationResult;

public record ConnectedSubgraphNetworkCreationResult(
		P2PNetwork createdNetwork,
		HashMap<String, String> nodeIdToNodeTemplateIdMapping) implements P2PNetworkCreationResult {

	@Override
	public P2PNetwork getCreatedNetwork() {
		return createdNetwork;
	}

}

File: org.palladiosimulator.blockchainsystems.threesim/src/main/kotlin/org/palladiosimulator/blockchainsystems/threesim/creation/connectedsubgraphnetwork/ConnectedSubgraphNetworkCreationResult.kt
package org.palladiosimulator.blockchainsystems.threesim.creation.connectedsubgraphnetwork

import org.palladiosimulator.blockchainsystems.core.system.abstractions.P2PNetwork
import org.palladiosimulator.blockchainsystems.core.system.abstractions.P2PNetworkCreationResult

data class ConnectedSubgraphNetworkCreationResult(
  val createdNetwork: P2PNetwork,
  val nodeIdToNodeTemplateIdMapping: HashMap<String, String>
) : P2PNetworkCreationResult {
  override fun getCreatedNetwork(): P2PNetwork {
    return createdNetwork
  }
}